dataplane/dpdk.h

```diff
@@ -0,0 +1,82 @@
+#pragma once
+#include <optional>
+#include <string>
+
+#include <rte_ring.h>
+
+#include "common/type.h"
+
+namespace dpdk
+{
+template<typename T>
+class Ring
```

Comment:
Why is this a template class? We're not using any specialization except for `rte_mbuf*` anyway